File: Java/level_order_traversal.java
//https://leetcode.com/problems/binary-tree-level-order-traversal/submissions/

public class levelordertraversal {
    public List<List<Integer>> levelOrder(TreeNode root) {
        
        if(root==null)
        {
            return new ArrayList<>();
        }
        TreeNode s=new TreeNode(Integer.MAX_VALUE);
        ArrayDeque<TreeNode> q=new ArrayDeque<>();
        //boolean vist[]=new boolean[]
        q.add(s);
        q.add(root);
        ArrayList<Integer> al=new ArrayList<>();
        List<List<Integer>> ans=new ArrayList<>();
        while(q.size()>0)
        {
            TreeNode top=q.remove();
            if(top==s)
            {
                if(q.size()>0){
                q.add(s);}
                if(al.size()>0)
                {
                    ans.add(al);
                    al=new ArrayList<>();
                }
            }
            else
            {
                al.add(top.val);
                
                if(top.left!=null) q.add(top.left);
                if(top.right!=null) q.add(top.right);
                
            }
        }
        
        return ans;
    }
}

File: CPP/Level_Order_Traversal.cpp
/**
 * Definition for binary tree
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode(int x) : val(x), left(NULL), right(NULL) {}
 * };
 */
vector<vector<int> > Solution::levelOrder(TreeNode* A) {
    vector<vector<int>> ans;
    if(A==NULL)
    return ans;
    queue<TreeNode*> q;
    q.push(A);
    while(!q.empty())
    {
        int n=q.size();
        vector<int> level;
        for(int i=0;i<n;i++)
        {
        TreeNode *curr=q.front();
        q.pop();
        if(curr->left)
        q.push(curr->left);
        if(curr->right)
        q.push(curr->right);
        level.push_back(curr->val);
        }
      ans.push_back(level);
    } 
    return ans;
}
